### PR TITLE
Stacked bar chart flex and legends

### PR DESCRIPTION
**Pull request by nulikartik**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Making stacked bar chart flex, so that user need not pass width attribute. The chart will flex accordingly based upon the available space. Consuming the new legends component with overflow behavior in the stacked bar chart. Making changes to Dashboard card to support changes to the stacked bar chart.
image
image

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 1]
```diff
@@ -1,90 +1,96 @@
...
+        {this._renderLegend && <Legends legends={legends} />}
       </div>
     );
   }
 
-  private _adjustProps = (): void => {
-    const { theme, className, styles, width, barHeight, legendData } = this.props;
-    const { palette } = theme!;
+  private _legendClicked(): void {
+    alert('Legend clicked');
```
Comment (Raghurk) [inline packages/charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx]: > alert('Legend clicked'); [](start = 3, length = 25)

do we need this alert. this does not seem to example

File: packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx [hunk 2]
```diff
@@ -1,113 +1,108 @@
...
-    const legendDataItems: ILegendDataItem[] = data.map((point: IDataPoint, index: number) => {
-      return {
-        legendText: point.x,
-        legendColor: this._colors[index % this._colors.length]
-      };
-    });
-
-    return <Legend renderData={legendDataItems} />;
+  private _legendClickAction(): void {
+    alert('legend clicked');
```
Comment (Raghurk) [inline packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx]: > alert('legend clicked'); [](start = 3, length = 25)

do we need this?

File: packages/charting/src/types/IDataPoint.ts [hunk 3]
```diff
@@ -11,3 +11,34 @@ export interface IDataPoint {
...
+   * Legend text for the datapoint in the chart
+   */
+  legend?: string;
+
+  /**
+   * data the datapoint in the chart
+   */
+  data?: number;
+
+  onClick?: VoidFunction;
```
[the remark this answers is not in the document]
Comment (nulikartik) [reply packages/charting/src/types/IDataPoint.ts]: added comment

File: packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx (hunk 2 above)
Comment (nulikartik) [reply packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx]: removed it, was placed as stub for hovercard behaviour we are going to implement next

File: packages/charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx (hunk 1 above)
Comment (nulikartik) [reply packages/charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx]: removed it

**Pull request merged**